chain/gen/slashfilter/slashfilter.go

```diff
@@ -48,7 +72,7 @@ func (f *SlashFilter) MinedBlock(bh *types.BlockHeader, parentEpoch abi.ChainEpo
 		}
 	}
 
-	{
+	if checkParentGrinding {
```

Comment:
Ok I'm starting to get it this is not two blocks in the epoch it is parent grinding.  I still need to understand what's going on but my first impression is we either need to change this logic to more closely match the actual slashing logic directly and if we can't there's something unsafe going on.

Reply:
Yes, we should make sure to be safer, because the above is less likely to happen. But what I understand is that in this case, the abnormal parent block is not introduced into the chain, so it will not be punished. I understand that it may be over-checked

Reply:
The nuance here is that the bad parent block is spread to the wider network.  I understand that it doesn't make it onto the main chain.  However that is not related to whether the block can cause a slashing event.  As long as the block is sent out over the network anyone else who records that block has the right to slash you if you also mine the second block which ignores the bad parent.

Reply:
Oh, so it is, and if so, isn't it a bit over-punishing?

Reply:
This also means that if miner get the right to manufacture two consecutive blocks, as long as the first block is invalid, the second block will also be lost.